### DISCUSSION TITLE
cli command to create database

### DISCUSSION THREAD

**Discussion by jesseduffield** [SeaQL/sea-orm#883] [General]
`sea-orm-cli migrate fresh` is able to (re)create all tables but doesn't actually create a database if one is not already present. Would it be possible to add that functionality in to sea-orm's CLI? I'm coming from rails background where there's a command to create the database itself. If it's deemed inappropriate to add such functionality to this crate, it would be nice to be clear in the docs that it's your own responsibility to create the database before you use sea-orm. Thanks.

**Comment by tyt2y3** [comment]
Okay what's the rails command indeed? I am thinking may be we can detect the absence of the database and offers a prompt to create that:

```sh
The database `my_db` does not exsit. Do you want to create it? Y/N
```

Although, that'd be the first case of interactive prompt in sea-orm-cli and so it might take some effort to make this happen

**Reply by jesseduffield** [reply]
As per the rails migration [docs](https://guides.rubyonrails.org/active_record_migrations.html):

> [4.2 Setup the Database](https://guides.rubyonrails.org/active_record_migrations.html#setup-the-database)
> The `bin/rails db:setup` command will create the database, load the schema, and initialize it with the seed data.
>
> [4.3 Resetting the Database](https://guides.rubyonrails.org/active_record_migrations.html#resetting-the-database)
> The `bin/rails db:reset` command will drop the database and set it up again. This is functionally equivalent to bin/rails db:drop db:setup.

I'm not sure we need to make the creation of the database interactive: I would have a command which always creates the database if not present, and perhaps if the user wants to skip that step we could have some other command for that. My assumption would be that the user would typically want the creation of the database to be handled by sea-orm.

Also, for what it's worth, this issue https://github.com/SeaQL/sea-orm/issues/765 would be resolved by having a process of dropping and recreating the database when running the `fresh` CLI command, because there would be no chance of having anything that the command neglected to delete.

**Reply by billy1624** [reply]
Hey @jesseduffield, thanks for the suggestions!! I think we can add a `setup` command for the CLI which create the database and schema (for PostgreSQL). And, the `fresh` command will delete the database and then run `setup` command.

But I'm thinking what will `setup` and `fresh` command do on SQLite? It's special because it has no such thing as database. It's just a file that contains data.

**Reply by jesseduffield** [reply]
Good question: I'm not well-acquainted with how things work with SQLite however rails also supports an SQLite backend so I'd expect there to be some public code somewhere showing how to go about it

**Reply by JonasCir** [reply]
I used https://docs.rs/rusqlite/latest/rusqlite/struct.Connection.html#method.open in one of my projects, it creates a working, empty DB at the given path if none exists.

**Comment by epipheus** [comment]
`touch` the file maybe?
